### DISCUSSION TITLE
Multicores

### DISCUSSION THREAD

**Discussion by johdcmlaker2000** [tetrane/reven-community#22] [Q&A] [answered]
Is there any way to start image with non default 1 processor and 1 core?
I tryed to set such lines in "Override custom options":

-smp 1,cores=4,sockets=4,threads=4

-smp cpus=4,cores=4,sockets=4,threads=4

-smp cpus=4,maxcpus=8

-smp 1,cores=4

etc and got all time ```An error occured during the start of the snapshot: The qemu session started, but was terminated early```

**Comment by dureuill** [comment, marked as answer]
I had some success with `Override custom options` set to e.g. `-smp 4`, on the Debian Stretch VM that we provide (`htop` does report 4 cores), with KVM enabled (disabling KVM prevents the VM from booting), and by restarting the VM from a disk snapshot (rather than loading a live snapshot, which would be the equivalent of hot-swapping the number of cores on a running VM).

Please note that REVEN does not currently support recording VMs with multiple cores, though.

I hope this helps!
